include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,782 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+ /*!\file
+  * \author Marie Hoffmann <marie.hoffmann AT fu-berlin.de>
+  * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+  * \brief Contains gap decorator to annotate sequences with gaps using a set.
+  */
+
+#pragma once
+
+#include <limits>
+#include <set>
+#include <tuple>
+#include <type_traits>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/std/view/view_all.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3
+{
+
+/*!\brief A gap decorator allows the annotation of sequences with gap symbols
+ *        while leaving the underlying sequence unmodified.
+ * \tparam inner_type The type of range that will be decorated with gaps; must model std::ranges::RandomAccessRange
+ *                    and std::ranges::SizedRange.
+ * \implements seqan3::aligned_sequence_concept
+ * \ingroup decorator
+ *
+ * \details
+ *
+ * This class may be used whenever you want to store or compute an alignment. The underlying (ungapped) sequence
+ * remains unmodified, and is augmented with gap information. The seqan3::gap_decorator_anchor_set behaves just like a
+ * vector over a gapped alphabet when iterating over it, inserting/erasing gaps or accessing a position. The only
+ * difference lies in the performance and size overhead (see below).
+ *
+ *
+ * ### Performance
+ *
+ * **n** The length of the underlying sequence.
+ * **k** The number of contiguous gaps (not gap symbols).
+ * **l** The total number of gap symbols.
+ *
+ * |            | access next | random access    | gap insert/erase at end | gap insert/erase random | size overhead |
+ * |------------|-------------|----------------- |-------------------------|-------------------------|---------------|
+ * | decorator  | \f$O(1)\f$  | \f$O(\log(k))\f$ | \f$O(\log(k))\f$        | \f$O(k)\f$              | O(k)          |
+ * | vector     | \f$O(1)\f$  | \f$O(1)\f$       | \f$O(n)\f$              | \f$O(n)\f$              | O(n)          |
```

Comment:
latex escapes around the last column's entries

Reply:
I still insist that insert at end of a vector is amortised constant. That's the point of a vector.